src/julia/install.sh

```diff
@@ -52,7 +53,10 @@ export DEBIAN_FRONTEND=noninteractive
 cleanup_apt
 check_packages curl ca-certificates
 
-su "${USERNAME}" -c "curl -fsSL https://install.julialang.org | sh -s -- --yes --default-channel ${CHANNEL}"
+su "${USERNAME}" -c "curl -fsSL https://install.julialang.org | sh -s -- --yes"
+for channel in "${channels[@]}"; do
+    su "${USERNAME}" -c "$(getent passwd "$USERNAME" | cut -d: -f6)/.juliaup/bin/juliaup add $channel"
```

Comment:
Perhaps you are struggling with variable expansion.
If you don't want to have variable expansion here, I think you can get around it by using single quotes instead of double quotes.

Also, you can use the special `_REMOTE_USER_HOME` env var.

https://containers.dev/implementors/features/#user-env-var